### ISSUE TITLE
Custom map stuff

### ISSUE THREAD

**Issue by peturfp** [Facepunch/rust-issues#20]
DONE - Base terrain
DONE - New terrain textures
DONE - Update terrain shader for custom maps
DONE - Terrain painting tools
DONE - River layout
DONE - Autospawn clutter palettes for each splat
DONE - Update rocks and cliffs
DONE - Biome atmospheric transition script
WIP - Updated trees and bushes
WIP - Sculpt in small terrain features, and fix elevation around the rivers
WIP - Spawn trees, ores, player spawnpoints.
SOON - Setup the barrel spawner

--- Mininum playable state? ---

SOON - Road
SOON - Spawn radtowns & monuments
SOON - River spawn polish
SOON - Proper waterfall & effects
SOON - Cloud effects around the distant mountains
SOON - General terrain sculpt & paint polish pass

MAYBE - Adjecent island?

**Comment by garrynewman** [comment]
ETA on being playable?

**Comment by peturfp** [comment]
Approx end of the month, if the upcoming new tools from Diogo speed up spawning/sculpting as much as I'm hoping.

But it will only be a rough.

**Comment by garrynewman** [comment]
I mean more in terms of the stuff that is required for it to be playable, rather than the island design. If we were to put this in tomorrow - why would it be unplayable?

**Comment by peturfp** [comment]
If we absolutely wanted to push this in now, I would have to spray random trees and rocks onto the flat areas that haven't been microsculpted yet.

By microsculpting, I mean turning relatively flat (boring) areas into something that offers more interesting gameaplay and cover. http://i.imgur.com/JwnRjc5.png


I mean, yes, we could do that, but I would only end up having to remove most of it again, because the spawned trees and rocks obviously break when I dramatically pull and push the terrain underneath.

To me it makes more sense to finish the microsculpting, so that all the trees and rocks from there on can be relatively permanent. I'm expecting it to take a week or so, despite the fact that it turns out we can't effectively improve Unity's poor terrain sculpting tools.

**Comment by garrynewman** [comment]
So in terms of AI, tree (re)spawning, ore (re)spawning, player spawnpoints, climate - we're all good right? 

These are the things that are important to me, the gameplay stuff - because without that it's just a pretty map, it's not playable.

**Comment by peturfp** [comment]
Climate yes, spawnpoints yes, resource respawning yes.

Haven't looked into AI yet. I guess maybe it needs a navmesh or something?

**Comment by garrynewman** [comment]
AI should just work as far as I know, as long as they spawn. Is AI not spawning yet?

**Comment by peturfp** [comment]
Yeah Helk just told me how to spawn them. So that should be good too.

**Comment by garrynewman** [comment]
New ETA on being playable?
